### PR TITLE
chore: Remove bump upper bound on fsspec requirement

### PR DESCRIPTION
**Pull request by cburroughs**
Alternative to #4461

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,12 +69,12 @@ lock-python-dependencies:
...
+	./pixi run --environment py310 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile --system --no-strip-extras setup.py --output-file sdk/python/requirements/py3.10-requirements.txt" > sdk/python/requirements/py3.10-requirements.txt
+	./pixi run --environment py310 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py3.10-ci-requirements.txt" > sdk/python/requirements/py3.10-ci-requirements.txt
+	./pixi run --environment py311 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile --system --no-strip-extras setup.py --output-file sdk/python/requirements/py3.11-requirements.txt" > sdk/python/requirements/py3.11-requirements.txt
+	./pixi run --environment py311 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py3.11-ci-requirements.txt" > sdk/python/requirements/py3.11-ci-requirements.txt
```
Comment (tokoko) [inline Makefile:77]: Why this change? Were you unable to install pixi?

Comment (cburroughs) [reply Makefile]: Sorry about the stray commit; removed now.

**Review by tokoko — APPROVED** [review]
lgtm
